Source/SwiftLintFramework/Rules/OperatorUsageWhitespaceRule.swift

```diff
@@ -78,10 +100,61 @@ public struct OperatorUsageWhitespaceRule: OptInRule, ConfigurationProviderRule
         let kinds = SyntaxKind.commentAndStringKinds()
 
         return file.matchPattern(pattern, excludingSyntaxKinds: kinds,
-                                 excludingPattern: excludingPattern).map {
-            return StyleViolation(ruleDescription: type(of: self).description,
-                                  severity: configuration.severity,
-                                  location: Location(file: file, characterOffset: $0.location))
+                                 excludingPattern: excludingPattern).flatMap {
+
+            let spacesPattern = oneSpace + "*"
+            let rangeRegex = NSRegularExpression
+                .forcePattern(spacesPattern + rangePattern + spacesPattern)
+
+            // if it's a range operator, the correction shouldn't have spaces
+            if let range = rangeRegex.firstMatch(in: file.contents,
+                                                 options: [], range: $0)?.range {
+                let correction = file.contents.bridge().substring(with: range)
+                    .trimmingCharacters(in: .whitespaces)
+                return (range, correction)
+            }
+
+            let pattern = spacesPattern + operators + spacesPattern
+            let operatorsRegex = NSRegularExpression.forcePattern(pattern)
+
+            guard let range = operatorsRegex.firstMatch(in: file.contents,
+                                                        options: [], range: $0)?.range else {
+                return nil
+            }
+
+            let operatorContent = file.contents.bridge().substring(with: range)
+                .trimmingCharacters(in: .whitespaces)
+            let correction = " " + operatorContent + " "
+
+            return (range, correction)
+        }
+    }
+
+    public func correctFile(_ file: File) -> [Correction] {
+        let violatingRanges = violationRanges(file: file).filter { range, _ in
+            return !file.ruleEnabledViolatingRanges([range], forRule: self).isEmpty
```

Comment:
why is `ruleEnabledViolatingRanges` called for every violation range? Couldn't you just do this?

```swift
let violatingRanges = file.ruleEnabledViolatingRanges(violationRanges(file: file), forRule: self)
```